src/plugins/data_source_management/server/saved_objects/data_source.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { SavedObjectsType } from 'opensearch-dashboards/server';
+
+export const dataSource: SavedObjectsType = {
```

Comment:
Shall we put data source saved object definition under `src/plugins/data/server/saved_objects`?

Reply:
What's the benefit to introduce data source saved object dependency to data plugin?

Reply:
> What's the benefit to introduce data source saved object dependency to data plugin?

If I'm not mistaken, the only server-side logic is to register the saved object type. We can remove the entire server side plugin by registering data sources in data plugin. I recall @zengyan-amazon making similar comments in previous PRs.

Reply:
Didn't see any benefit, I won't approve to add to data plugin I'm convinced with data point.

Reply:
> Didn't see any benefit, I won't approve to add to data plugin unless I'm convinced with data point.

Data points FYI: Index pattern is defined in https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/data/server/saved_objects instead of index pattern management plugin

Reply:
> Could you explain why it is good idea? Or we have to do this? This against single responsibility principle

I don't think it's a blocking comment, nor it against single responsibility principle. One of the benefits is to remove entire server side plugin and simplify code structure.

Reply:
introduce new dependency to data plugin is huge concern to me. Leave data source here, it will only be registered when feature is enabled. There is no single line of code change need to make in data plugin. 

I'm ok to create another server side code only plugin. But that's overkill.

Reply:
> introduce new dependency to data plugin is huge concern to me. Leave data source here, it will only be registered when feature is enabled. There is no single line of code change need to make in data plugin.
> 
> I'm ok to create another server side code only plugin. But that's overkill.

I think we are aligned that server side plugin is an overkill. Instead of saying it introduces a dependency on the data plugin, I would say creating a new service in the data plugin and allowing other plugins to rely on it is not a bad idea. We can publish the current code and leave the community to review.

Reply:
Adding another plugin is overkill. For now. Leave that in DSM plugin is not.

Adding DS to data plugin will pollute the data plugin, it is bad idea to me. Even if we have to that means data plugin is design in a bad way which doesn't allow extension without touch its code.

Reply:
> Adding another plugin is overkill. For now. Leave that in DSM plugin is not.
> 
> Adding DS to data plugin will pollute the data plugin, it is bad idea to me. Even if we have to do that means data plugin is design in a bad way which doesn't allow extension without touch its code.

https://github.com/opensearch-project/OpenSearch-Dashboards/pull/2052